src/storage/mvcc/reader/point_getter.rs

```diff
@@ -223,94 +223,102 @@ impl<S: Snapshot> PointGetter<S> {
             }
 
-            match write.write_type {
-                WriteType::Put => {
-                    self.statistics.write.processed_keys += 1;
-
-                    if self.omit_value {
-                        return Ok(Some(vec![]));
+            loop {
+                // We may seek to another key. In this case, it means we cannot find the specified key.
+                {
+                    let cursor_key = self.write_cursor.key(&mut self.statistics.write);
+                    if !Key::is_user_key_eq(cursor_key, user_key.as_encoded().as_slice()) {
+                        return Ok(None);
                     }
-                    match write.short_value {
-                        Some(value) => {
-                            // Value is carried in `write`.
-                            return Ok(Some(value.to_vec()));
+                }
+
+                let write = WriteRef::parse(self.write_cursor.value(&mut self.statistics.write))?;
+
+                if !write.check_gc_fence_as_latest_version(self.ts) {
+                    return Ok(None);
+                }
+
+                match write.write_type {
+                    WriteType::Put => {
+                        self.statistics.write.processed_keys += 1;
+
+                        if self.omit_value {
+                            return Ok(Some(vec![]));
                         }
-                        None => {
-                            let start_ts = write.start_ts;
-                            return Ok(Some(self.load_data_from_default_cf(start_ts, user_key)?));
+                        match write.short_value {
+                            Some(value) => {
+                                // Value is carried in `write`.
+                                return Ok(Some(value.to_vec()));
+                            }
+                            None => {
+                                let start_ts = write.start_ts;
+                                return Ok(Some(
+                                    self.load_data_from_default_cf(start_ts, user_key)?,
+                                ));
+                            }
                         }
                     }
+                    WriteType::Delete => {
+                        return Ok(None);
+                    }
+                    WriteType::Lock | WriteType::Rollback => {
+                        // Continue iterate next `write`.
+                    }
                 }
-                WriteType::Delete => {
+
+                if !self.write_cursor.next(&mut self.statistics.write) {
                     return Ok(None);
                 }
-                WriteType::Lock | WriteType::Rollback => {
-                    // Continue iterate next `write`.
-                }
             }
-
-            if !self.write_cursor.next(&mut self.statistics.write) {
-                return Ok(None);
-            }
-        }
+        };
+        let timer = Instant::now();
```

Comment:
How about providing a wrap function for this function, to avoid too many diff? Otherwise you will suffer a lot from cherry pick.

Reply:
I think the `timing` marco at [minstant](https://github.com/tikv/minstant/blob/master/minstant_macro/src/lib.rs) is suitable for this case.